File: 3779 minimum-number-of-operations-to-have-distinct-elements/solution.java
class Solution {
    public int minOperations(int[] nums) {
        Set<Integer> set = new HashSet<>();
        int idx = -1;
        for(int i=nums.length-1;i>=0;i--) {
            if(!set.add(nums[i])) {
                idx = i;
                break;
            }
        }
        return idx == -1 ? 0 : (idx + 3)/3;
    }
}

File: 3779 minimum-number-of-operations-to-have-distinct-elements/solution.rs
use std::collections::HashSet;
impl Solution {
    pub fn min_operations(nums: Vec<i32>) -> i32 {
        let mut set = HashSet::new();
        let mut idx = -1;
        let n = nums.len();
        for i in (0..n).rev() {
            if !set.insert(nums[i]) {
                idx = i as i32;
                break;
            }
        }
        if(idx==-1) {
            return 0
        }
        (idx+3)/3
    }
}
